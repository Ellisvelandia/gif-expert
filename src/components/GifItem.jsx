import React from "react";
import PropTypes from "prop-types";

const GifItem = ({ title, url }) => {
  return (
    <div className="card">
      <img src={url} alt={title} loading="lazy" />
      <p>{title}</p>
    </div>
  );
};

GifItem.prototype = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default GifItem;
