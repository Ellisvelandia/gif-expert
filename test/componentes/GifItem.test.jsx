import { render, screen } from "@testing-library/react";
import GifItem from "../../src/components/GifItem";
describe("Test GifItems", () => {
  const title = "Saitama";
  const url = "https://one-punch.com/saitama.jpg";
  test("should match with snapshot", () => {
    const { container } = render(<GifItem title={title} ulr={url} />);
    expect(container).toMatchSnapshot();
  });

  test("should be show image , URL and ALT", () => {
    render(<GifItem title={title} ulr={url} />);
    // const { src, alt } = screen.getByRole("img");
    // expect(src).toBe(url);
    // expect(alt).toBe(alt);

    expect(screen.getByRole("img").src);
  });

  test("should be show the title in component", () => {
    render(<GifItem title={title} ulr={url} />);
    expect(screen.getAllByText(title)).toBeTruthy();
  });
});
