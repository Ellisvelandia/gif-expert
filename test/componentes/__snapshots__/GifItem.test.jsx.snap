// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifItems should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      loading="lazy"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
